src/modelmanager.cpp

```diff
@@ -543,6 +605,36 @@ Status ModelManager::reloadModelWithVersions(ModelConfig& config) {
     std::shared_ptr<model_versions_t> versionsToRetire;
 
     auto model = getModelIfExistCreateElse(config.getName());
+
+    // first set custom loader name to empty string so that any changes to name can be captured
+    std:: string emptyString = std::string();
+    model->setCustomLoaderName(emptyString);
+
+    if (config.isCustomLoaderRequiredToLoadModel()) {
+        custom_loader_options_config_t customLoaderOptionsConfig = config.getCustomLoaderOptionsConfigMap();
+        const std::string loaderName = customLoaderOptionsConfig["loader_name"];
+
+        auto& customloaders = ovms::CustomLoaders::instance();
+        auto loaderPtr = customloaders.find(loaderName);
+        if (loaderPtr != nullptr) {
+            spdlog::info("Custom Loader to be used : {}", loaderName);
+            model->setCustomLoaderName(loaderName);
```

Comment:
dispose setCustomLoaderName and checking for it. Use explicitly ModelConfig & if really needed create helper function for checking if custom loader is required, what is name etc.

Reply:
Yes. Doing that. However there no access to the config at model level. It has access at modelinstance level. So, except in model, removed it every where. We need loader name in model to retire the model